### PR TITLE
Orion/create release 8

### PR DESCRIPTION
**Pull request by borland**
feat: release create gains outputformat json and basic
feat: If a release version is tied to a donor package, lock that as the release version and just let them add +metadata
feat: fancy help screen for package question loop
feat: can reset package loop
fix: uses git ref canonical name when addressing version controlled projects

### CODE REVIEW

File: pkg/cmd/release/create/create.go [hunk 1]
```diff
@@ -69,6 +72,47 @@ const (
...
+var packageOverrideLoopHelpText = heredoc.Doc(`
+bold(PACKAGE SELECTION)
+ This screen presents the list of packages used by your project, and the steps
+ which reference them. 
+ If an item is dimmed (gray text) this indicates that the attribute is duplicated.
+ For example if you reference the same package in two steps, the second will be dimmed. 
+
+bold(COMMANDS)
+ Any any point, you can enter one of the following:
+ - green(?) to access this help screen
```
Comment (borland) [inline pkg/cmd/release/create/create.go:84]: This is like the world's shittiest version of markdown, but the "bang-for-buck" we get with a handful of regexes is great.
FYI @domenicsim1 @jbristowe if you want to re-use the prettifyHelp function in other places

File: pkg/cmd/release/create/create.go [hunk 2]
```diff
@@ -908,31 +975,31 @@ func AskQuestions(octopus *octopusApiClient.Client, stdout io.Writer, asker ques
 				return fmt.Errorf("internal error: can't find donor package in deployment process template - version controlled configuration file in an invalid state")
 			}
 
-			defaultNextVersion = donorPackage.Version
+			versionMetadata, err := askVersionMetadata(asker, donorPackage.PackageID, donorPackage.Version)
```
Comment (borland) [inline pkg/cmd/release/create/create.go:978]: @slewis74 merry christmas

File: pkg/cmd/release/create/create.go [hunk 3]
```diff
@@ -1066,15 +1138,40 @@ outer_loop:
...
+		return "", err
+	}
+	return result, nil
+}
+
+func askVersionMetadata(ask question.Asker, packageId string, packageVersion string) (string, error) {
+	var result string
+	if err := ask(&survey.Input{
+		Default: "",
+		Message: fmt.Sprintf("Using release version %s from package %s. Add +metadata? (blank for none)", packageVersion, packageId),
```
Comment (borland) [inline pkg/cmd/release/create/create.go:1155]: TODO: I struggled to come up with a sensible (but not too long) prompt here.
The root problem is that I expect statistically zero people to understand what version metadata is
We should discuss this with the team.

File: pkg/cmd/release/create/create_test.go [hunk 4]
```diff
@@ -1153,7 +1307,71 @@ func TestReleaseCreate_AutomationMode(t *testing.T) {
 			_, err = testutil.ReceivePair(cmdReceiver)
 			assert.Nil(t, err)
 
-			assert.Equal(t, "Successfully created release version 1.2.3 (Releases-999) using channel Alpha channel (Channels-32)\n", stdOut.String())
+			assert.Equal(t, heredoc.Doc(`
+				Successfully created release version 1.2.3 using channel Alpha channel
```
Comment (borland) [inline pkg/cmd/release/create/create_test.go:1311]: Note I took out the ID's so there's no more Releases-999 and Channels-32

File: pkg/cmd/release/list/list.go [hunk 5]
```diff
@@ -135,6 +135,7 @@ func listRun(cmd *cobra.Command, f factory.Factory, flags *ListFlags) error {
 
 	return output.PrintArray(allReleases, cmd, output.Mappers[ReleaseViewModel]{
 		Json: func(item ReleaseViewModel) any {
+			// TODO should the ReleaseID go in the JSON as well as the release number? How about channelID?
```
Comment (borland) [inline pkg/cmd/release/list/list.go:138]: I'd like to discuss this.

We've decided to steer away from ID's, and now the "human readable" output just says the channel name and release version string. -f basic just says the version string.

But I feel like the JSON should still contain the ID. 🔥🔥🔥??
At the moment it is halfway housed, we get the channel ID, but not the release ID

(Note this applies to both `release list` and `release create`, they output the same structure)

File: pkg/cmd/release/create/create.go [hunk 6]
```diff
@@ -1042,20 +1109,25 @@ outer_loop:
...
-		}
-
-		if resolvedOverride != nil {
+		} else if answer == "r" { // reset! All the way back to the calculated versions, discarding even the stuff that came in from the cmdline
+			if len(packageVersionOverrides) > 0 {
+				packageVersionOverrides = make([]*PackageVersionOverride, 0)
+				overriddenPackageVersions = ApplyPackageOverrides(packageVersionBaseline, packageVersionOverrides)
+			}
+			continue // print table and go again
+		} else if resolvedOverride != nil {
```
Comment (domenicsim1) [inline pkg/cmd/release/create/create.go:1130]: This is starting to get bit big, maybe time to use a switch instead. They don’t have problems like falling through in go.

File: pkg/cmd/release/create/create.go (hunk 3 above)
Comment (domenicsim1) [reply pkg/cmd/release/create/create.go:1155]: Don’t mind the length too much. Got any ideas in how we could make it shorter @slewis74 ?

**Review by domenicsim1 — APPROVED** [review]

File: pkg/cmd/release/create/create.go (hunk 6 above)
Comment (borland) [reply pkg/cmd/release/create/create.go:1130]: I did that internally, but that it changes the semantics of `break` within the 'yes' section... so I'd need an outer break with a label... and then it's not clear that's an improvement anymore

Comment (borland) [reply pkg/cmd/release/create/create.go:1130]: actually turns out we already had an outer break label, so the penalty doesn't apply; I've changed to a switch now and it's better, thanks

**Pull request merged**